strawberry/enum.py

```diff
@@ -102,6 +102,12 @@ def _process_enum(
             deprecation_reason = item_value.deprecation_reason
             item_value = item_value.value
 
+            # TODO: this check should always be true, but if it's not we should
+            # raise an error (???)
+
+            if item.name not in cls._value2member_map_:
+                cls._value2member_map_[item_value] = item
```

Comment:
@devkral I think here we can use item.value, and that should fix your use case as well, will try it soon 😊

Reply:
this should work. But it is still unclean (you access internals).

What speaks against my solution?

It is much simpler and doesn't changes the internals of Enum.


Hint: you can change the item value by changing the item._value_ attribute not by changing the item.value

Reply:
> this should work. But it is still unclean (you access internals) 

I think this is fine, I wanted to treat `strawberry.enum_value` similar to how enum.auto is treated.

> and you still have somewhere the references to the former Enum item (memory) floating around.

We need that for the type generation, no? :)

> What speaks against my solution?
> It is much simpler and doesn't changes the internals of Enum.

Your solution is good too, but I think I prefer making use of internals to completely hide the fact that we are using strawberry.enum_value. The enum should be a python enum without any additional things. Similar to what happens when we use strawberry.field :)

> Hint: you can change the item value by changing the item._value_ attribute not by changing the item.value

Oh man! Thanks so much for this, I didn't manage to find a proper solution so far! :D